libs/alsa-lib/patches/300-musl-alsa-lib-stdint.patch

```diff
@@ -0,0 +1,170 @@
+--- a/include/pcm.h
++++ b/include/pcm.h
+@@ -33,6 +33,7 @@
+ extern "C" {
+ #endif
+ 
++#include <stdint.h>
+ /**
+  *  \defgroup PCM PCM Interface
+  *  See the \ref pcm page for more details.
+@@ -1108,10 +1109,10 @@ int snd_pcm_format_width(snd_pcm_format_
+ int snd_pcm_format_physical_width(snd_pcm_format_t format);		/* in bits */
+ snd_pcm_format_t snd_pcm_build_linear_format(int width, int pwidth, int unsignd, int big_endian);
+ ssize_t snd_pcm_format_size(snd_pcm_format_t format, size_t samples);
+-u_int8_t snd_pcm_format_silence(snd_pcm_format_t format);
+-u_int16_t snd_pcm_format_silence_16(snd_pcm_format_t format);
+-u_int32_t snd_pcm_format_silence_32(snd_pcm_format_t format);
+-u_int64_t snd_pcm_format_silence_64(snd_pcm_format_t format);
++uint8_t snd_pcm_format_silence(snd_pcm_format_t format);
```

Comment:
@pprindeville - Inclusion of`<sys/types.h>` is actually done by `asoundlib.h` which, in turn, includes `pcm.h`. However, the app source which is including `asoundlib.h` may have already included `<sys/types.h>` so it is the individual apps which need to define `_GNU_SOURCE`. This change, which I admit is kind of messy, makes it possible to not have to change every app using alsa-lib in this manner. For example: by simply adding `-D_GNU_SOURCE` to TARGET_CPPFLAGS in the **mpg123** Makefile instead does make this change unnecessary.